easybuild/easyblocks/p/parmetis.py

```diff
@@ -167,19 +168,31 @@ def install_step(self):
         # and header files in the Lib directory (capital L). The following symlink are hence created.
         try:
             llibdir = os.path.join(self.installdir, 'Lib')
+            if os.path.lexists(llibdir):
+                os.remove(llibdir)
             os.symlink(libdir, llibdir)
             for f in ['metis.h', 'parmetis.h']:
+                if os.path.lexists(os.path.join(libdir, f)):
+                    os.remove(os.path.join(libdir, f))
                 os.symlink(os.path.join(includedir, f), os.path.join(libdir, f))
         except OSError as err:
             raise EasyBuildError("Something went wrong during symlink creation: %s", err)
 
     def sanity_check_step(self):
         """Custom sanity check for ParMETIS."""
 
-        custom_paths = {
+        if '-DSHARED=1' in self.cfg['configopts']:
+            shlib_ext = get_shared_lib_ext()
+            custom_paths = {
+                        'files': ['include/%smetis.h' % x for x in ["", "par"]] +
+                                 ['lib/libparmetis.%s' % shlib_ext],
+                        'dirs': ['Lib']
+                       }
+        else:
+            custom_paths = {
                         'files': ['include/%smetis.h' % x for x in ["", "par"]] +
                                  ['lib/lib%smetis.a' % x for x in ["", "par"]],
```

Comment:
@lexming Since only the library extension is different, we can do better...

In addition, you're only checking for `libparmetis.so`, not `libmetis.so`?

Something like this:

```python
if '-DSHARED=1' in self.cfg['configopts']:
    lib_ext = get_shared_lib_ext()
else:
    lib_ext = 'a'

custom_paths = {
    'files': ['include/%smetis.h' % x for x in ['', 'par']] +
             ['lib/lib%smetis.%s' % (x, lib_ext) for x in ['', 'par']],
}
```

This is not perfect though, since there are probably other ways to enable building shared libraries (`-DSHARED=ON` for example)...

Reply:
Unfortunately ParMETIS builds `libparmetis.so` and `libmetis.a` with `SHARED=1`. Nonetheless, I simplified the code setting the library paths in `custom_paths` and also added a check on `SHARED` that considers all possible `true` strings in CMake.